src/AutoRest.CSharp/Common/Output/Models/Types/SignatureType.cs

```diff
@@ -0,0 +1,247 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis;
+using AutoRest.CSharp.Input.Source;
+using AutoRest.CSharp.Mgmt.AutoRest;
+using AutoRest.CSharp.Output.Models.Shared;
+using System;
+using System.Linq;
+using AutoRest.CSharp.Mgmt.Decorator;
+using System.Diagnostics.CodeAnalysis;
+
+namespace AutoRest.CSharp.Output.Models.Types
+{
+    internal class SignatureType
+    {
+        private readonly string _defaultNamespace;
+        private readonly string _defaultName;
+        private readonly SignatureType? _customization;
+        private readonly SignatureType? _previousContract;
+
+        // Missing means the method with the same name is missing from the current contract
+        // Updated means the method with the same name is updated in the current contract, and the list contains the previous method and current methods including overload ones
+        private readonly (IReadOnlyList<MethodSignature> Missing, IReadOnlyList<(List<MethodSignature> Current, MethodSignature Previous)> Updated)? _methodChangeset;
```

Comment:
since this is quite often used inside this class, could we make a struct or record to represent the type here?
It could make it more readable.

Reply:
Updated